infrastructure/core/network.tf

```diff
@@ -77,3 +77,46 @@ resource "azurerm_private_dns_zone_virtual_network_link" "all" {
     azurerm_private_dns_zone.all
   ]
 }
+
+resource "azurerm_network_security_group" "core" {
+  name                = "nsg-default-${var.naming_suffix}"
+  location            = azurerm_resource_group.core.location
+  resource_group_name = azurerm_resource_group.core.name
+
+  security_rule {
+    name                       = "deny-internet-outbound-override"
+    description                = "Blocks outbound internet traffic unless an explicit outbound-allow rule exists. Overrides the default rule 65001"
+    priority                   = 2000
+    access                     = "Deny"
+    protocol                   = "*"
+    direction                  = "Outbound"
+    destination_address_prefix = "Internet"
+    destination_port_range     = 443
+    source_address_prefix      = "*"
+    source_port_range          = "*"
+  }
+}
+
+# Network watcher must be added in real-data scenarios, thus will fail if network_watcher is undefined
```

Comment:
Worth a precondition for better failure handling if the config isn't set correctly?